system.go

```diff
@@ -68,3 +69,82 @@ func StartEtcd(nodes []vagrantssh.TestbedNode) error {
 
 	return nil
 }
+
+//ServiceStart starts a systemd service unit
+func ServiceStart(n vagrantssh.TestbedNode, srv string) (string, error) {
+	return n.RunCommandWithOutput(fmt.Sprintf("sudo systemctl start %s", srv))
+}
+
+//ServiceStatus queries and returns status result of systemd service unit
+func ServiceStatus(n vagrantssh.TestbedNode, srv string) (string, error) {
+	return n.RunCommandWithOutput(fmt.Sprintf("systemctl status %s", srv))
+}
+
+// WaitForDone polls for checkDoneFn function to return true up until specified timeout
+func WaitForDone(doneFn func() (string, bool), timeoutSec int, timeoutMsg string) (string, error) {
+	tick := time.Tick(time.Duration(2) * time.Second)
+	timeout := time.After(time.Duration(timeoutSec) * time.Second)
+	doneCount := 0
+	for {
+		select {
+		case <-tick:
+			if ctxt, done := doneFn(); done {
+				doneCount++
+				// add some resilliency to poll inorder to avoid false positives,
+				// while polling more frequently
+				if doneCount == 2 {
+					// end poll
+					return ctxt, nil
+				}
+			}
+			//continue polling
+		case <-timeout:
+			ctxt, done := doneFn()
+			if !done {
+				return ctxt, fmt.Errorf("wait timeout. Msg: %s", timeoutMsg)
+			}
+			return ctxt, nil
+		}
+	}
+}
+
+func serviceActionAndWaitForState(n vagrantssh.TestbedNode, srv string, timeoutSec int, state string,
```

Comment:
ah, k..

yeah, right now the only consumers of this were in the same file, so I didn't export the symbol. But thinking more, it can be used outside as well. I will make the change.